lua/tests/gmod/classes/convar/__tostring.lua

```diff
@@ -0,0 +1,35 @@
+function GetTestConVar() -- Abusing the loading order :>
+	return CreateConVar( "ConVarTest", "1.2", 0, "This is a Test", -10, 10 )
+end
+
+return {
+    groupName = "ConVar:__tostring",
+
+    cases = {
+        {
+            name = "Exists on the ConVar meta table",
+            func = function()
+                local meta = FindMetaTable( "ConVar" )
+                expect( meta.__tostring ).to.beA( "function" )
+            end
+        },
+
+        --[[{ -- NOTE: __tostring will error if given nil self? :/ (Try it again but in a shutdown hook which maybe runs after Lua created ConVars were destroyed?)
+            name = "Returns the right value",
+            func = function()
+                local a = ConVar( "NoneExistentConVar" )
+
+                expect( a:__tostring() ).to.equal( "ConVar [NULL]" )
+            end
+        },]]
```

Comment:
You can test this behavior!

```suggestion
        {
            name = "Errors when called with a nil self",
            func = function()
                local convar_tostring = FindMetaTable( "ConVar" ).__tostring

                local function test()
                    convar_tostring( nil )
                end

                expect( test ).to.errWith( [[bad argument #1 to '__tostring' (ConVar expected, got nil)]] )
            end
        },
```